testing/agent-exporter/src/main/java/io/opentelemetry/javaagent/testing/util/KeysVerifyingPropagator.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.testing.util;
+
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.propagation.TextMapGetter;
+import io.opentelemetry.context.propagation.TextMapPropagator;
+import io.opentelemetry.context.propagation.TextMapSetter;
+import java.util.Collection;
+import javax.annotation.Nullable;
+
+/**
+ * A TextMapPropagator wrapper that verifies TextMapGetter methods can be called without errors.
+ *
+ * <p>This propagator wraps another propagator and calls methods during extraction. This catches
+ * compatibility issues where underlying library APIs change (e.g., NoSuchMethodError).
+ *
+ * <p>Note: This is a copy of the class in testing-common to avoid dependency conflicts
```

Comment:
when I tried adding test-common as a dependency, I hit a bunch of ASM errors. I was able to get it to work by adding some dependency exclusions, but this seemed simpler. I can go back to the other approach if that is preferable